docs/reference/template-tags.md

```diff
@@ -0,0 +1,200 @@
+
+#### `areaName`
+
+The name (a string) of the area field as defined in the field schema.
+
+#### `contextOptions` (optional)
+
+The context options object is added after area tag arguments following the `with` keyword. It is an object with keys matching the names of widget types allowed in that particular area field. Each key is assigned a value that will be passed into the widget template as `data.contextOptions`. See the [image widget guide](/guide/core-widgets.md#image-widget) for an example.
+
+Context options are optional for all core and official Apostrophe widget types.
+
+::: note
+Context options are not the best place for most widget configuration. That should be done in the [area field configuration](/reference/field-types/area.md#widgets). Context options are used to override that with options that only apply to the specific template context.
+:::
+
+## `component`
+
+The `component` tag inserts an asynchronous component into the template. See the [async components guide](/guide/async-components.md) for more on using this feature.
+
+### Usage
+
+```django
+{% component 'module:componentName' with data %}
+```
+
+**Example:**
+```django
+{% component 'product:newest' with { max: 3 } %}
+```
+
+### Arguments
+
+#### `module:componentName`
+
+The primary argument is a combination of the name of a module name and the name of an async component from that module, separated by a colon. All async components belong to a specific module, though multiple modules may have components with the same name.
+
+#### `data` (optional)
+
+The data argument, following the `with` keyword, is available in the async component template as `data`. It can be any data type, however it is a best practice to use an object with subproperties.
+
+## `fragment`
+
+The `fragment` tag *declares* a template fragment that will be inserted elsewhere with [`render`](#render) or [`rendercall`](#rendercall). See the [template fragments guide](/guide/fragments.md) for more on using this feature.
+
+This tag must be closed with an `endfragment` tag.
+
+### Usage
+
+```django
+{% fragment name(parameters) %}
+  {# Fragment markup #}
+{% endfragment %}
+```
+
+**Examples:**
+```django
+{% fragment button(text, action, options = {}) %}
+  <button class="o-button {{ options.class }}" data-action="{{ action }}">
+    {{ text }}
+  </button>
+{% endfragment %}
+```
+
+```django
+{% fragment card(data = {}) %}
+  <section class="o-card">
+    <h2>{{ data.heading }}</h2>
+    <div>
+      {# `rendercaller` connects to the `rendercall` tag below. #}
+      {{ rendercaller() }}
+    </div>
+  </section>
+{% endfragment %}
+```
+
+### Arguments
+
+#### `name(parameters)`
+
+The fragment name used to reference it in `render` and `rendercall` tags. It should be written as a function with parentheses. The parentheses may include argument names if needed, which would be referenced in variable brackets.
```

Comment:
How about: "It should be written *like* a function declaration, with parentheses."

I don't understand "which would be referenced in variable brackets," I suggest just removing that.

Since this is reference you should probably add "parameters should be separated by commas and may optionally include a default value assignment."